Create a sentinel file for each dynamo test failure

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120451
* __->__ #120355

Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

cc @albanD

oulgen committed:
Create a sentinal file for each dynamo test failure

[ghstack-poisoned]

aorenste reviewed (approved):
LGTM. Maybe consider splitting into two diffs - one that creates all the new files and a separate one that changes the dynamo_test_failures.py. That might make it a little easier for people who hit this as a merge conflict...

oulgen committed:
Update on "Create a sentinal file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```



[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

clee2000 commented on torch/testing/_internal/dynamo_test_failures.py:
```diff
@@ -1,4 +1,7 @@
 # mypy: ignore-errors
+import os
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
```
Disclaimer that I'm not very experienced with this

This file is located in torch, so `__file__` refers to where ever the torch pip package is installed.  Locally this usually works because we do an editable install, but in CI, we build the package and install that normally, so its in site-packages or something like that.  If you want it to show up in the pip package I think it needs to be part of the build somehow, but I'm not sure that would be a good idea 

Some solutions might be:
1. generate a file or something similar prior to calling anything from torch (probably in run_tests.py) that can then be read by common_utils  
2. inspect the stack to get the calling file (the actual test file located in test/, like test/test_ops.py, which is not located in torch) and somehow get back to this folder to generate the list

Both of these solutions have their pros and cons and will probably need to be aware of where the base directory is (in CI I think its test/) and will probably require some path string parsing weirdness

aorenste replied:
What about teaching setup.py (or torchgen, etc) to build a file list based on the directory structure? I assume we have some generated files that end up in the pip wheel in a similar fashion. That way the merging is workable but the loaded file is "just python".

oulgen committed:
Update on "Create a sentinal file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

[ghstack-poisoned]

oulgen committed:
Update on "Create a sentinel file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

[ghstack-poisoned]

zou3519 commented on test/torch_np/numpy_tests/core/test_multiarray.py:
```diff
@@ -3613,8 +3613,9 @@ def test_ip_types(self):
-    def test_byteorder(self, dtype):
+    @parametrize("dtype", (True, False))
+    def test_byteorder(self, greater):
+        dtype = ">i4" if greater else "<i4"
```
Could you also fix/rewrite scripts/compile_tests/update_failures.py to work with the new format?

oulgen committed:
Update on "Create a sentinel file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

oulgen committed:
Update on "Create a sentinel file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

[ghstack-poisoned]

oulgen committed:
Update on "Create a sentinel file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

oulgen committed:
Update on "Create a sentinel file for each dynamo test failure"


Created via
```
import os
current_dir = os.path.dirname(os.path.abspath(__file__))
directory = os.path.join(current_dir, 'dynamo_expected_failures')
for name in dynamo_expected_failures:
    path = os.path.join(directory, name)
    with open(path, 'w') as fp:
        pass
```

Differential Revision: [D54036062](https://our.internmc.facebook.com/intern/diff/D54036062)

cc albanD

[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54036062).

oulgen commented:
@pytorchbot merge

oulgen commented:
@pytorchbot merge
